docs/src/content/docs/fr/components/code.mdx

```diff
@@ -0,0 +1,103 @@
+---
+title: Code
+description: Apprenez à afficher du code avec coloration syntaxique dans Starlight sans blocs de code Markdown.
+---
+
+import { Code } from '@astrojs/starlight/components';
+
+Le composant `<Code>` affiche du code avec coloration syntaxique.
+Il est utile lorsque l'utilisation d'un [bloc de code Markdown](/fr/guides/authoring-content/#blocs-de-code) n'est pas possible, par exemple pour afficher des données provenant de sources externes comme des fichiers, des bases de données ou des API.
+
+import Preview from '~/components/component-preview.astro';
+
+<Preview>
+
+<Code
+	slot="preview"
+	code={`## Bienvenue
+
+Bonjour depuis **l'espace**!`}
+lang="md"
+title="exemple.md"
+ins={3}
+/>
+
+</Preview>
+
+## Import
+
+```tsx
+import { Code } from '@astrojs/starlight/components';
+```
+
+## Utilisation
+
+Utilisez le composant `<Code>` pour afficher du code avec coloration syntaxique, par exemple lorsque vous affichez du code provenant de sources externes.
+
+Consultez la [documentation sur le « Composant Code » d'Expressive Code](https://expressive-code.com/key-features/code-component/) pour plus de détails sur l'utilisation du composant `<Code>` et la liste des props disponibles.
+
+<Preview>
+
+```mdx
+import { Code } from '@astrojs/starlight/components';
+
+export const exampleCode = `console.log("Cela peut provenir d'un fichier ou d'un CMS !");`;
+export const fileName = 'exemple.js';
+export const highlights = ['fichier', 'CMS'];
+
+<Code code={exampleCode} lang="js" title={fileName} mark={highlights} />
+```
+
+<Fragment slot="markdoc">
+
+```markdoc
+{% code
+	 code="console.log(\"Cela peut provenir d'un fichier ou d'un CMS !\");"
```

Comment:
Slightly different syntax compared to the English version to handle escaping `"` but I tested it locally to make sure it works.